src/server/scripts/EasternKingdoms/TheStockade/boss_hogger.cpp

```diff
@@ -98,7 +100,7 @@ class boss_hogger : public CreatureScript
             if (me->HasUnitState(UNIT_STATE_CASTING))
                 return;
 
-            while (uint32 eventId = events.ExecuteEvent())
+            if (uint32 eventId = events.ExecuteEvent())
```

Comment:
Could you comment this change a bit more?

Reply:
Don't cast 2 spells at same update (if it somehow does happen), because of casting check above

Reply:
Those functions redundancy can be reduced a lot by using the BossAI::ExecuteEvent hook.
The whole UpdateAI function could be replaced with
```
void ExecuteEvent(uint32 eventId) override
{
    switch (eventId)
    {
    case EVENT_VICIOUS_SLICE:
        DoCastVictim(SPELL_VICIOUS_SLICE);
        events.Repeat(Seconds(10), Seconds(14));
        break;
     case EVENT_MADDENING_CALL:
        DoCast(SPELL_MADDENING_CALL);
        events.Repeat(Seconds(15), Seconds(20));
        break;
    }
}
```